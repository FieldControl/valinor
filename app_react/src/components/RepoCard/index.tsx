import { RepoCardProps } from "../@interfaces/IRepoCard";
import { formatDistanceToNow } from 'date-fns';

export default function RepoCard({
  full_name,
  owner,
  description,
  updated_at,
  watchers_count,
  language,
  topics,
  html_url
}: RepoCardProps) {
  // Converte a string de data para um objeto Date
  const updatedAtDate = new Date(updated_at);

  // Formata a data em extenso
  const formattedUpdatedAt = formatDistanceToNow(updatedAtDate, { addSuffix: true });

  return (
    <a href={html_url} title="Ir para o repositório" target="_blank">
        <div className="mb-4 w-full text-white m-auto block p-6 border border-gray-500 rounded-lg shadow dark:bg-gray-800 dark:border-gray-700 dark:hover:bg-gray-700">
      <h5 className="flex flex-col sm:flex-row gap-3 text-white mb-2 text-2xl font-bold tracking-tight">
        <img src={owner.avatar_url} className="rounded-full mb-2 sm:mb-0" width="30" alt="Avatar" />
        <span>{full_name}</span>
      </h5>
      <p className="font-normal text-white dark:text-gray-400">
        {description}
      </p>
      <ul className="flex flex-wrap gap-2 pt-4">
        {topics.map((item, index) => (
          <li className="rounded-lg bg-blue-900 p-2" key={index}>
            {item}
          </li>
        ))}
      </ul>
      <div className="flex flex-col sm:flex-row justify-between items-center text-gray-500 mt-4">
        <p className="mb-2 sm:mb-0">Ultima atualização: {formattedUpdatedAt}</p>
        <p className="mb-2 sm:mb-0">Stars: {watchers_count}</p>
        <p>{language}</p>
      </div>
    </div>
    </a>
  
  );
}
