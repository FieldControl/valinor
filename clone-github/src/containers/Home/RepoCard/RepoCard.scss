.repoCard {
  box-sizing: border-box;
  padding: 20px;
  position: relative;
  display: flex;
  flex-direction: column;
  align-items: left;
  text-align: left;
  width: 100%;
  height: 100%;
  border-top: 1px solid silver;
}

.repoCard__link {
  & em {
    font-weight: bold;
  }
  &:hover {
    opacity: 0.7;
  }
}
.repoCard__content {
  box-sizing: border-box;
  padding: 1px;
  font-size: 12px;
  color: #586069;
  width: 100%;
  list-style: none;
  display: flex;
  justify-content: space-between;
  align-items: center;
  flex-wrap: wrap;
  & li {
    margin-top: 5px;
  }
}
.repoCard__description {
  box-sizing: border-box;
  padding: 1px;
  font-size: 14px;
  color: #24292e;
  margin-top: 5px;
}

.icon {
  width: 14px;
  height: 14px;
  margin-right: 3px;
  position: relative;
}
.icon--repo {
  width: 16px;
  height: 16px;
  position: absolute;
  left: 0;
}
.repoCardIcon {
  display: flex;
  align-items: center;
}
.icon__circle {
  box-sizing: border-box;
  width: 12px;
  height: 12px;
  border-radius: 50%;
  margin-right: 3px;
  margin-left: 3px;
  border: 2px solid black;
}
