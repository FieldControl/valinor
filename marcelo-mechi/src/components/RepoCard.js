function RepoCard() {
    return (
        <div>RepoCard</div>
    )
}

export default RepoCard;