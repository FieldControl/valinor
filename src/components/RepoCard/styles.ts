import styled, { css } from 'styled-components';
import { RiStarLine } from 'react-icons/ri';
import { Repositories, Fork } from '../../styles/Icons';

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  padding: 16px;
  border: 1px solid var(--border);
  border-radius: 6px;
`;

export const Topside = styled.div`
  > header {
    display: flex;
    align-items: center;

    > a {
      margin-left: 8px;
      font-size: 14px;
      font-weight: 600;
      color: var(--link);

      text-decoration: none;

      &:focus,
      &:hover {
        text-decoration: underline;
      }
    }
  }

  > p {
    margin: 8px 0 16px;
    font-size: 12px;
    color: var(--gray);
    letter-spacing: 0.1px;
    line-height: 20px;
    > a {
      text-decoration: none;
    }
  }
`;

const iconCSS = css`
  width: 16px;
  height: 16px;
  fill: var(--icon);
  flex-shrink: 0;
`;

export const RepoIcon = styled(Repositories)`
  ${iconCSS}
`;

export const Botside = styled.div`
  > ul {
    display: flex;
    align-items: center;

    > li {
      display: flex;
      align-items: center;
      margin-right: 16px;

      > span {
        margin-left: 5px;
        font-size: 12px;
        color: var(--gray);
      }
    }
  }
`;

export const LanguageDot = styled.div`
  width: 12px;
  height: 12px;
  border-radius: 50%;
  flex-shrink: 0;
  background: ${props => (props.color ? props.color : '#8257e5')};
`;

export const StarIcon = styled(RiStarLine)`
  ${iconCSS}
`;

export const ForkIcon = styled(Fork)`
  ${iconCSS}
`;
