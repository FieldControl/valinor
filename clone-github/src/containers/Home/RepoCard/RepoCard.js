import React from "react";
import PropTypes from "prop-types";
import "./RepoCard.scss";
import starIcon from "../../../assets/star.png";
import repoIcon from "../../../assets/repo.png";
import { formatNumber, formatString } from "../../../helpers/format";

const RepoCard = ({
  full_name,
  description,
  updated_at,
  language,
  open_issues,
  stargazers_count,
  license,
}) => {
  return (
    <div className="repo__card">
      <a className="repo__card__link repo__card__icon" href="/">
        {" "}
        <img
          width="12px"
          height="12px"
          className="icon icon--repo"
          src={repoIcon}
        />
        {full_name}
      </a>
      {description ? (
        <p className="repo__card__description">{formatString(description)}</p>
      ) : null}

      <ul className="repo__card__content">
        <li className="repo__card__icon">
          <img width="12px" height="12px" className="icon" src={starIcon} />
          {formatNumber(stargazers_count)}
        </li>
        {language ? (
          <li className="repo__card__icon">
            {" "}
            <span className="icon__circle"></span>
            {language}
          </li>
        ) : null}
        {license && license !== null ? (
          <li>{license.key.toUpperCase()} license</li>
        ) : null}
        <li>Updated {new Date(updated_at).toLocaleDateString()}</li>
        <li>{open_issues} issues need help</li>
      </ul>
    </div>
  );
};

RepoCard.propTypes = {
  full_name: PropTypes.string,
  description: PropTypes.string,
  updated_at: PropTypes.string,
  language: PropTypes.string,
  open_issues: PropTypes.number,
  stargazers_count: PropTypes.number,
  license: PropTypes.object,
};

export default RepoCard;
