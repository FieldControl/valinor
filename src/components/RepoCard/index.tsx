/* eslint-disable react/jsx-one-expression-per-line */
/* eslint-disable react/jsx-indent */
import React from 'react';
import { Link } from 'react-router-dom';
import { Tooltip } from '@material-ui/core';

import languageColors from '../../utils/language-colors';
import kFormatter from '../../utils/kFormatter';

import { useStyles } from '../../styles/MaterialUI';
import { Container, Topside, RepoIcon, Botside, StarIcon, ForkIcon, LanguageDot } from './styles';

interface Props {
  username: string;
  reponame: string;
  description: string;
  language: string;
  stars: number;
  forks: number;
  isForked: boolean;
}

const RepoCard: React.FC<Props> = ({ username, reponame, description, language, stars, forks, isForked }) => {
  const classes = useStyles();

  const languageName = language ? language.replace(' ', '-').toLowerCase() : 'other';
  const languageColor = languageColors[languageName];
  return (
    <Container>
      <Topside>
        <header>
          {isForked ? <ForkIcon /> : <RepoIcon />}
          <Tooltip title={`Go to ${reponame}`} placement="bottom" arrow classes={{ tooltip: classes.tooltip }}>
            <Link to={`/${username}/${reponame}`}>{reponame}</Link>
          </Tooltip>
        </header>
        <p>{description}</p>
      </Topside>

      <Botside>
        <ul>
          {language && (
            <Tooltip title={`Repository main language: ${language}`} placement="bottom" arrow classes={{ tooltip: classes.tooltip }}>
              <li>
                <LanguageDot color={languageColor || '#8257e5'} />
                <span>{language}</span>
              </li>
            </Tooltip>
          )}
          {stars && stars > 0 ? (
            <Tooltip title={`Repository has ${stars} stars`} placement="bottom" arrow classes={{ tooltip: classes.tooltip }}>
              <li>
                <StarIcon />
                <span>{kFormatter(stars)}</span>
              </li>
            </Tooltip>
          ) : null}
          {forks && forks > 0 ? (
            <Tooltip title={`Repository was forked by ${forks} users`} placement="bottom" arrow classes={{ tooltip: classes.tooltip }}>
              <li>
                <ForkIcon />
                <span>{kFormatter(forks)}</span>
              </li>
            </Tooltip>
          ) : null}
        </ul>
      </Botside>
    </Container>
  );
};

export default RepoCard;
