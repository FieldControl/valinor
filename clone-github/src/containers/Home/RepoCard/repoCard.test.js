import { render, cleanup } from "@testing-library/react";
import RepoCard from "./RepoCard.js";

afterEach(cleanup);

test("should take a snapshoot", () => {
  const { asFragment } = render(<RepoCard />);
  expect(asFragment(<RepoCard />)).toMatchSnapshot();
});
