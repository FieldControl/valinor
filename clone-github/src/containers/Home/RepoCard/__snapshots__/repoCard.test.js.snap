// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshoot 1`] = `
<DocumentFragment>
  <div
    class="repo__card"
  >
    <a
      class="repo__card__link repo__card__icon"
      href="/"
    >
       
      <img
        class="icon icon--repo"
        height="12px"
        src="repo.png"
        width="12px"
      />
    </a>
    <ul
      class="repo__card__content"
    >
      <li
        class="repo__card__icon"
      >
        <img
          class="icon"
          height="12px"
          src="star.png"
          width="12px"
        />
        und.em
      </li>
      <li>
        Updated Invalid Date
      </li>
      <li>
         issues need help
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
